client/src/features/settings/Settings.tsx

```diff
@@ -1,7 +1,225 @@
-import { PageTitle } from '@common';
+import { PageTitle, ActionButtons } from '@common';
+import { Form, Formik } from 'formik';
+import * as Yup from 'yup';
+import { useDarkMode } from '@hooks/useDarkMode';
+import { InputText } from 'primereact/inputtext';
+import { Password } from 'primereact/password';
+import { Dropdown } from 'primereact/dropdown';
+
+interface SettingsFormValues {
+	theme: string;
+	username: string;
+	password: string;
+	confirmPassword: string;
+	language: string;
+}
+
+const languages = [
+	{ label: 'English', value: 'en' },
+	{ label: 'Spanish', value: 'es' },
+	{ label: 'French', value: 'fr' },
+	{ label: 'German', value: 'de' },
+];
+
+const themes = [
+	{ label: 'Light', value: 'light' },
+	{ label: 'Dark', value: 'dark' },
+];
+
+const settingsValidationSchema = Yup.object().shape({
+	theme: Yup.string().required('Theme setting is required'),
+	username: Yup.string().required('Username is required').min(3, 'Username must be at least 3 characters'),
+	password: Yup.string()
+		.min(8, 'Password must be at least 8 characters')
+		.matches(/[0-9]/, 'Password must contain at least one number')
+		.matches(/[a-z]/, 'Password must contain at least one lowercase letter')
+		.matches(/[A-Z]/, 'Password must contain at least one uppercase letter'),
+	confirmPassword: Yup.string()
+		.oneOf([Yup.ref('password')], 'Passwords must match')
+		.required('Confirm Password is required'),
+	language: Yup.string().required('Language is required'),
+});
+
 
 const Settings = () => {
-	return <PageTitle>Settings</PageTitle>;
+	const [theme, toggleTheme] = useDarkMode();
+
+	const initialValues: SettingsFormValues = {
+		theme,
+		username: '',
+		password: '',
+		confirmPassword: '',
+		language: 'en',
+	};
+
+	const handleSubmit = (values: SettingsFormValues) => {
+		if (values.theme !== theme) {
+			toggleTheme();
+		}
+		// Handle other settings updates here
+		console.log('Settings updated:', values);
+	};
+
+	return (
+		<div className="max-w-3xl mx-auto px-4">
+			<PageTitle>Settings</PageTitle>
+
+			<Formik
+				initialValues={initialValues}
+				validationSchema={settingsValidationSchema}
+				onSubmit={handleSubmit}
+				enableReinitialize
+			>
+				{({ submitForm, resetForm, values, errors, touched, handleChange, setFieldValue }) => (
+					<>
+						<Form className="mt-6">
+							<div className="card p-4">
+								<div className="flex flex-col gap-6">
+									<div className="flex-1">
+										<div className="flex flex-col gap-2">
+											<label htmlFor="theme">Theme</label>
+											<Dropdown
+												id="theme"
+												name="theme"
+												value={theme}
+												options={themes}
+												onChange={(e) => {
+													setFieldValue('theme', e.value);
+													toggleTheme();
+												}}
+												className={`w-full ${touched.theme && errors.theme ? 'p-invalid' : ''}`}
+											/>
+											{touched.theme && errors.theme && (
```

Comment:
I think we can remove the Theme required validation. Since we just have two themes, it doesn't make sense to use a Dropdown but a input switch like this: https://primereact.org/inputswitch/ 
So when the button is on, it is like the lights are turned on. That would be True for Light Mode.

Reply:
That being said. Replace the Dropdown for the InputSwitch component